config/configv2.go

```diff
@@ -84,25 +84,30 @@ const (
 const typeAndNameSeparator = "/"
 
 // Load loads a Config from Viper.
-func Load(v *viper.Viper) (*configmodels.Config, error) {
+func Load(
+	v *viper.Viper,
+	receiverFactories map[string]receiver.Factory,
```

Comment:
Is it worth defining a struct for the factory map triplet? A lot of code does nothing but get the 3 maps for factories and pass to another function, e.g.
```go
type FactoryMaps struct {
  receiverFactories map[string]receiver.Factory
  processorFactories map[string]processor.Factory
  exporterFactories map[string]exporter.Factory
}
```

Reply:
It sometimes passes just one of those down to individual components instead of passing all three all the time. I guess it would be useful but at current usage I don't see that big of a difference. 

I was however thinking of moving the factory interface definitions to config module. Config highly depends on it and is the biggest (only?) consumer of the factories anyway. Defining the interfaces in config module would completely eliminate config's dependency on exporter, receiver and processor modules. I didn't look too deep into this and didn't want to club it with the current PR but we can discuss it for another PR if you want.